core/src/main/java/org/elasticsearch/gateway/GatewayAllocator.java

```diff
@@ -87,9 +83,8 @@ public GatewayAllocator(Settings settings, TransportNodesListGatewayStartedShard
         logger.debug("using initial_shards [{}]", initialShards);
     }
 
-    public void setReallocation(final ClusterService clusterService, final AllocationService allocationService) {
-        this.clusterService = clusterService;
-        this.allocationService = allocationService;
+    public void setReallocation(final ClusterService clusterService, final RoutingService routingService) {
```

Comment:
I also wonder if we can trash this method then altogether a make `GatewayAllocator implement ClusterStateListener` and add it to the `ClusterService` once constructed or maybe inside the ClusterService...